core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -615,6 +615,12 @@ package object config {
     .stringConf
     .createOptional
 
+  private[spark] val METRICS_STATIC_SOURCES_ENABLED =
+    ConfigBuilder("spark.metrics.static.sources.enabled")
```

Comment:
Could you explain why they are static and the others are not static?

Reply:
Apache Spark defines like the following.
```
private[spark] object StaticSources {
  /**
   * The set of all static sources. These sources may be reported to from any class, including
   * static classes, without requiring reference to a SparkEnv.
   */
  val allSources = Seq(CodegenMetrics, HiveCatalogMetrics)
}
```

Reply:
`spark.metrics.staticSources.enabled` might look better. Using static as the name space looks weird. cc @cloud-fan @Ngone51

Reply:
`staticSources` sounds better.

Reply:
I agree that staticsSources sounds better.
BTW there is probably room for a naming convention for the "switch parameters" for enabling/disabling metrics? Currently we have:  
`spark.metrics.static.sources.enabled`  
`spark.app.status.metrics.enabled`  
`spark.sql.streaming.metricsEnabled`

Reply:
See also #26692